### ISSUE TITLE
custom set test relies on equality assumptions

### ISSUE THREAD

**Issue by macintux** [exercism/erlang#298]
Using `?assertEqual` on the results of the set operations makes unwarranted assumptions:

* Erlang equality applies to the set representations
  * This assumption violates the "How it works internally doesn't matter, as long as it behaves like a set of unique elements" description
* Given the two likely set representations one might find that would support equality, lists and tuples, equality would still only work if the sets are always sorted in the same order, but there's no indication in the description that the implementation is supposed to be an ordered set

The tests that bit me:
* union_of_non_empty_sets_contains_all_unique_elements
* difference_of_two_non_empty_sets_is_a_set_of_elements_that_are_only_in_the_first_set
* add_to_non_empty_set

There may well be others that assume predictable ordering.

The logical fix would seem to be to convert all such tests from `?assertEqual` to applying the same logic as `sets_with_the_same_elements_are_equal_test`, i.e. rely on `custom_set:equal`.

I'm looking at https://github.com/exercism/erlang/blob/master/testgen/src/tgen_custom-set.erl but it's already midnight here so I may not file a PR any time soon.

**Comment by NobbZ** [comment]
Yeah, you are totally right, and as far as I remember we used `custom_set:equal/2` (or something equivalent) in the earlier handcrafted test suite for this exercise.

I'm not sure why I got this wrong in the generated testsuite.

But yes, the file you pointed at is the correct one.

I will wait for a PR up to the holidays and if nothing happens I will tackle this when I update canonical data between the years.
